src/test/java/com/uber/cadence/internal/replay/ReplaceDeciderDecisionTaskWithHistoryIteratorTest.java

```diff
@@ -0,0 +1,247 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.replay;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import com.uber.cadence.*;
+import com.uber.cadence.client.WorkflowClientOptions;
+import com.uber.cadence.internal.worker.SingleWorkerOptions;
+import com.uber.cadence.serviceclient.IWorkflowService;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.Duration;
+import java.util.*;
+
+import org.apache.thrift.TException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class ReplaceDeciderDecisionTaskWithHistoryIteratorTest {
+    @Mock
+    private IWorkflowService mockService;
+
+    @Mock
+    private DecisionContextImpl mockContext;
+
+    @Mock
+    private DecisionsHelper mockedHelper;
+
+    private static final int MAXIMUM_PAGE_SIZE = 10000;
+    private final String WORKFLOW_ID = "testWorkflowId";
+    private final String RUN_ID = "testRunId";
+    private final String DOMAIN = "testDomain";
+    private final String START_PAGE_TOKEN = "testPageToken";
+    private final WorkflowExecution WORKFLOW_EXECUTION =
+            new WorkflowExecution().setWorkflowId(WORKFLOW_ID).setRunId(RUN_ID);
+    private final HistoryEvent START_EVENT =
+            new HistoryEvent()
+                    .setWorkflowExecutionStartedEventAttributes(new WorkflowExecutionStartedEventAttributes())
+                    .setEventId(1);
+    private final History HISTORY = new History().setEvents(Collections.singletonList(START_EVENT));
+    private final PollForDecisionTaskResponse task =
+            new PollForDecisionTaskResponse()
+                    .setWorkflowExecution(WORKFLOW_EXECUTION)
+                    .setHistory(HISTORY)
+                    .setNextPageToken(START_PAGE_TOKEN.getBytes());
+
+    private Object iterator;
+
+    private void setupDecisionTaskWithHistoryIteratorImpl() {
+        try {
+            // Find the inner class first
+            Class<?> innerClass = findDecisionTaskWithHistoryIteratorImplClass();
+
+            // Get the constructor with the specific parameter types
+            Constructor<?> constructor =
+                    innerClass.getDeclaredConstructor(
+                            ReplayDecider.class, PollForDecisionTaskResponse.class, Duration.class);
+            constructor.setAccessible(true);
```

Comment:
I'd have a preference for modifying the class and annotating it as `@VisibleForTesting`.

Reply:
Good point. Changed to VisibleForTesting, it looks better then reflection hacks with setAccessible.